testing-suite/staking-v4/scenarios/PR_6114.py

```diff
@@ -0,0 +1,181 @@
+    assert "success" in add_blocks_until_tx_fully_executed(tx_hash)
+
+    # check that nodes are staked
+    for key in all_keys:
+        assert "staked" in key.get_status(_A.public_address())
+
+    # move 2 blocks before calling validator/statistics
+    assert "success" in add_blocks(2)
+    time.sleep(1)
+
+    # check that nodes are in auction
+    for key in all_keys:
+        assert "auction" in key.get_state()
+
+    # === STEP 2 ==============================================================
+    # 2) Add blocks and epochs until 1 of this 2 keys are eligible - let's call it epoch X
+
+    eligible_key = add_blocks_until_key_eligible(all_keys)
+
+    current_epoch = proxy_default.get_network_status().epoch_number
+    print("Eligible key: ", eligible_key.public_address())
+    print("Current epoch:", current_epoch)
+
+    # === STEP 3 ==============================================================
+    # 3) unStake the eligible key in epoch X
+
+    tx_hash = unStake(_A, eligible_key)
+
+    # check if tx is success
+    assert "success" in add_blocks_until_tx_fully_executed(tx_hash)
+
+    current_epoch = proxy_default.get_network_status().epoch_number
+    print("Key unstaked in epoch:", current_epoch)
+
+    # check if key is now unStaked on getBlsKeyStatus
+    assert "unStaked" in eligible_key.get_status(_A.public_address())
+
+    # eligible_key becomes un_staked_key to be easier to read the test
+    un_staked_key = eligible_key
+
+    # === STEP 4 ==============================================================
+    # 4) call auction list api after 2 blocks and:
+
+    # wait 1 sec and move to the last block of epoch before saving all qualified and unqualified keys
+    time.sleep(1)
+    assert "success" in add_blocks_until_last_block_of_current_epoch()
+
+    # 4.1) unStaked key status should be "eligible (leaving)" on validator/statistics
+    assert un_staked_key.get_state() == "eligible (leaving)"
+
+    epoch_x = proxy_default.get_network_status().epoch_number
+    print("Key is leaving in epoch:", epoch_x)
+
+    # 4.2) we should have still 8 keys qualified
+    qualified_keys_in_epoch_x = get_keys_from_validator_auction(QUALIFIED=True)
+    assert len(qualified_keys_in_epoch_x) == 8
+
+    # 4.3) we should have still 2 keys not qualified
+    not_qualified_keys_in_epoch_x = get_keys_from_validator_auction(QUALIFIED=False)
+    assert len(not_qualified_keys_in_epoch_x) == 2
+
+    # 4.4) we should have 39 keys with status "eligible"
+    eligible_keys_in_epoch_x = get_keys_from_validator_statistics(needed_state="eligible")
+    assert len(eligible_keys_in_epoch_x) == 39
+
+    # 4.5) we should have still 16 keys with status "waiting"
+    waiting_keys_in_epoch_x = get_keys_from_validator_statistics(needed_state="waiting")
+    assert len(waiting_keys_in_epoch_x) == 16
+
+    # === STEP 5 ==============================================================
+    # 5) go to epoch X+1  and 5 blocks and make sure:
```

Comment:
Make sure what?

Reply:
It is continuing with 5.1, 5.2 , checks that are related to Step 5.  Makes more sense at the top of the file where the scenario steps are.